### ISSUE TITLE
auto-gen-config no longer add Rails cops

### ISSUE THREAD

**Issue by graaff** [rubocop-hq/rubocop#4238]
With rubocop 0.48.1 the auto-gen-config option no longer adds Rails cops to the todo file. This also appeared to be the case with rubocop 0.48.0, but I could not test this well due to #4225 at the time.

Initially I thought this may be related to the new TargetRailsVersion option, but this happens both with and without this option in .rubocop.yml. I have tested it with "TargetRailsVersion: 4.2"

0.48.1 (using Parser 2.3.3.1, running on ruby 2.2.7 x86_64-linux)

**Comment by graaff** [comment]
I just tested with another internal Rails project and here the Rails cops are correctly added with auto-gen-config.

Any hints on how to debug this further?

**Comment by d-lebed** [comment]
Do you run it with -R key? Does .rubocop.yml in other project contain
```yaml
Rails:
  Enabled: true
```
?

**Comment by graaff** [comment]
Rails is enabled in both projects in .rubocop.yml. I normally don't use -R. However, using -R does fix the issue. With "rubocop -R --auto-gen-config" a todo file is created that includes the Rails cops, and afterwards "rubocop" (with or without -R) no longer shows failing cops. This is the .rubocop.yml for this project:

```yaml
inherit_from: .rubocop_todo.yml

Rails:
  Enabled: true

AllCops:
  TargetRubyVersion: 2.1
  Exclude:
    - db/schema.rb
    - features/**/*

Lint/UnreachableCode:
  Exclude:
    - db/migrate/**/*

Security/Eval:
  Exclude:
    - config/deploy.rb
    - lib/tasks/check.rake
    - script/**/*

Style/DoubleNegation:
  Enabled: false

Style/FileName:
  Exclude:
    - script/**/*
    - theme/**/*

Style/SpaceInsideStringInterpolation:
  Enabled: false
```

**Comment by jonas054** [comment]
Nope. I don't understand it and I cannot reproduce. Could it be that some other configuration file is involved? Run with `-d` to see which configuration files are read.

**Comment by Drenmi** [comment]
We are using Rails cops and `--auto-gen-config` at work, and I haven't seen this issue.

**Comment by graaff** [comment]
@jonas054 thanks! It turns out there was another .rubocop.yml file in a sub-directory that was getting picked up, as shown with -d. Excluding that directory fixes the issue for us.

The other .rubocop.yml is really basic and doesn't mention Rails either way, so having it influence this is a bit unexpected. Whether or not that is a bug I don't know. :smile:  Please close the issue if you feel this is expected behavior.

**Comment by jonas054** [comment]
I don't think it's expected. Can you provide some more details, e.g. the output when running with `-d`?

**Comment by tibbon** [comment]
I've encountered this too!

**Comment by mikegee** [comment]
@tibbon would please post the output when you run `rubocop -d`?

**Comment by bbatsov** [comment]
I'm closing this ticket due to no recent activity. Feel free to re-open it if you ever come back to it.
